### ISSUE TITLE
spawn ENAMETOOLONG error on discarding files

### ISSUE THREAD

**Issue by charblar** [desktop/desktop#2833]
### Description

Github Desktop still seems to have issues discarding a large number of files and will give a "spawn ENAMETOOLONG error". 

An issue was [already reported for this issue](https://github.com/desktop/desktop/issues/1668) but it was marked as resolved. 

### Version

**GitHub Desktop version:** 1.0.0

**OS version:** Windows 10 Home - Version 1703 - OS Build 15063.608

### Steps to Reproduce

![screenshot of issue](https://i.imgur.com/0SIxgjC.png)

1. Have a large amount of files with staged changes
2. Right click on the bar which states the number of staged files and click discard changes

**Expected behavior:** Discard changes to files

**Actual behavior:** spawn ENAMETOOLONG error

**Reproduces how often:** 100%
### Logs

[2017-09-22.desktop.production.txt](https://github.com/desktop/desktop/files/1323148/2017-09-22.desktop.production.txt)

**Comment by shiftkey** [comment]
@charblar thanks for bringing this to our attention. I can't see the specific error in the log, but I'll have a look at recreating this error on my end.

**Comment by charblar** [comment]
@shiftkey no problem, the app has come a long way that this is only a minor annoyance!

**Comment by shiftkey** [comment]
> Have a large amount of files with staged changes

Aha, this was where I ended up - I should have noticed it sooner.

I was able to reproduce this using an absurd little script to generate a large number of changed files in the Typescript repo from #1668, stage them and then try to discard all. It churns in the background while moving files to the recycle bin, but then _because the files are staged_ we run this `reset` while passing in the full array of staged paths:

https://github.com/desktop/desktop/blob/1a3d64c3429298ac14dad91413bef2b2e4291f38/app/src/lib/git/reset.ts#L77

I was able to tweak this to  succeed with a `git reset` of 32,767 characters, and fail for a `git reset` of 32,811 characters. This fits with the maximum length for a command being limited to 32,767 characters for a shell on Windows ([[citation]](https://blogs.msdn.microsoft.com/oldnewthing/20031210-00/?p=41553)).

Anyway, we should work out how many characters we can pass in here once you factor in the full path to `git.exe`, and transform `paths` into multiple arrays, if required.
